CMSgov/easi-app
# [EASI-2818] [EASI-2871] CEDAR Core v2.0.0 Migration

# EASI-2818 EASI-2871

## Changes and Description

### General Changes
- Introduced `CEDAR_CORE_API_VERSION`, an environment variable used to configure the path used when calling CEDAR Core. Acceptable values are `1.0.0` and `2.0.0`

### Changes to support differences in the CEDAR Core API v2.0.0
- Role assignee types were being returned with different casing than 1.0.0, so all comparisons to expected role type strings are now case insensitive
- Role type IDs (hard-coded on the frontend for organizing points of contact by their role) changed in v2, so a change was made to use the `roleTypeName` property instead, which is consistent across both v1 and v2


## How to test this change

- Update your `.envrc.local` to have `export CEDAR_CORE_API_VERSION=2.0.0`
- Run `direnv allow`
- Ensure the CMS Cloud VPN is running
- `scripts/dev up`
- Ensure that System Profile works as expected

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
